Core.ApplicationServices/KLEApplicationService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections.Generic;
+using Core.ApplicationServices.Authorization;
+using Core.ApplicationServices.Model.Result;
+using Core.DomainModel.KLE;
+using Core.DomainModel.Organization;
+using Core.DomainServices.Model.Result;
+using Core.DomainServices.Repositories.KLE;
+
+namespace Core.ApplicationServices
+{
+    public class KLEApplicationService : IKLEApplicationService
+    {
+        private readonly IOrganizationalUserContext _organizationalUserContext;
+        private readonly IKLEStandardRepository _kleStandardRepository;
+        private readonly IKLEUpdateHistoryItemRepository _kleUpdateHistoryItemRepository;
+
+        public KLEApplicationService(IOrganizationalUserContext organizationalUserContext,
+            IKLEStandardRepository kleStandardRepository,
+            IKLEUpdateHistoryItemRepository kleUpdateHistoryItemRepository)
+        {
+            _organizationalUserContext = organizationalUserContext;
+            _kleStandardRepository = kleStandardRepository;
+            _kleUpdateHistoryItemRepository = kleUpdateHistoryItemRepository;
+        }
+
+        public Result<KLEStatus, OperationFailure> GetKLEStatus()
+        {
+            if (!_organizationalUserContext.HasRole(OrganizationRole.GlobalAdmin))
+            {
+                return Result<KLEStatus, OperationFailure>.Failure(OperationFailure.Forbidden);
+            }
+            var lastUpdated = _kleUpdateHistoryItemRepository.GetLastUpdated();
+            return Result<KLEStatus, OperationFailure>.Success(_kleStandardRepository.GetKLEStatus(lastUpdated));
+        }
+
+        public Result<IEnumerable<KLEChange>, OperationFailure> GetKLEChangeSummary()
+        {
+            if (!_organizationalUserContext.HasRole(OrganizationRole.GlobalAdmin))
+            {
+                return Result<IEnumerable<KLEChange>, OperationFailure>.Failure(OperationFailure.Forbidden);
+            }
+            return Result<IEnumerable<KLEChange>, OperationFailure>.Success(_kleStandardRepository.GetKLEChangeSummary());
+        }
+
+        public Result<KLEUpdateStatus, OperationFailure> UpdateKLE()
+        {
+            if (!_organizationalUserContext.HasRole(OrganizationRole.GlobalAdmin))
+            {
+                return Result<KLEUpdateStatus, OperationFailure>.Failure(OperationFailure.Forbidden);
+            }
+
+            var publishedDate = _kleStandardRepository.UpdateKLE(_organizationalUserContext.UserId, _organizationalUserContext.ActiveOrganizationId);
```

Comment:
@strongmindsmhs  Hvad nu hvis jeg API mæssigt kalder den 2 gange. Forsøger den så at køre det hele igen?. Skal operationen udføres såfremt den allerede er up2date?

Reply:
@mrjsawdk Right. Har tilføjet ny test for dette tilfælde og modificeret koden til at validere status. I tilfældet ikke upToDate returneres fejl, "BadInput" (har af samme grund ændret navn på denne til "BadRequest", synes dette er mest dækkende)